lib/tree.rb

```diff
@@ -16,40 +16,168 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(Log n)
+  # Space Complexity: O(1)
   def add(key, value)
```

Comment:
👍 , but the space complexity is O(log n) because of the call stack, assuming the tree is balanced.